[PearlynnT] iP

PearlynnT opened the pull request:
# Violet
> "If you leave everything to the last minute, you only need a minute to do it."

[Click here to watch a video on procrastination](https://www.youtube.com/watch?v=arj7oStGLkU)

Violet is your _personal assistant chatbot_. It is ...
- text-based
- easy to use
- your **best friend** :cupid:

All you need to do is ...
1. run the code
2. add your tasks
3. let it manage your tasks for you :wink:

Here's the `main` method of the code:
```Java
public static void main(String[] args) {
    new Duke("data/duke.txt").run();
}
```

Features:
- [x] Managing tasks
- [x] Search task using keyword
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

PearlynnT committed:
Level-0

PearlynnT committed:
Level-1

PearlynnT committed:
Level-1

PearlynnT committed:
Level-2

PearlynnT committed:
Level-3

PearlynnT committed:
Level-4

PearlynnT committed:
A-TextUiTesting

PearlynnT committed:
Level-5

PearlynnT committed:
Level-5

PearlynnT committed:
Level-6

PearlynnT committed:
doc

PearlynnT committed:
Add exception classes

PearlynnT committed:
Add Level-7

PearlynnT committed:
Add list method

PearlynnT committed:
Merge branch-Level-7

PearlynnT committed:
Add Level-8

PearlynnT committed:
Merge branch-Level-8

PearlynnT committed:
Add A-MoreOOP

PearlynnT committed:
Merge branch-A-MoreOOP

PearlynnT committed:
Add A-Packages

PearlynnT committed:
Merge branch-A-packages

PearlynnT committed:
Add checkstyle

PearlynnT committed:
Merge add-gradle-support

PearlynnT committed:
Add A-Gradle

PearlynnT committed:
Add A-JUnit

PearlynnT committed:
Add A-Jar

PearlynnT committed:
Add JavaDoc comments

PearlynnT committed:
Tweak code to comply with coding standard

PearlynnT committed:
Add Level-9

PearlynnT committed:
Add Level-9

PearlynnT committed:
Merge branch-A-JavaDoc

PearlynnT committed:
branch-A-CodingStandard

PearlynnT committed:
Resolve merge conflict

PearlynnT committed:
Change import order

PearlynnT committed:
Merge branch-A-CheckStyle

jovkusuma commented on data/duke.txt:
```diff
@@ -0,0 +1,4 @@
+T | 1 | read book
+D | 0 | return book | 2023-06-06 1800
+E | 0 | project meeting | 2023-08-06 1400 - 2023-08-06 1600
+T | 1 | join sports club
```
Shouldn't this file be ignored?

jovkusuma commented on src/main/java/duke/util/Storage.java:
```diff
@@ -0,0 +1,109 @@
+ */
+
+public class Storage {
+    private final String pathname;
```
Variable names should be in camelCase.

```suggestion
    private final String pathName;
```

jovkusuma commented on src/main/java/duke/util/Ui.java:
```diff
@@ -0,0 +1,151 @@
+ */
+
+public class Ui {
+    private final String line = "\t____________________________________________________________";
```
Isn't this a constant? If it is, ~~line~~ LINE should be in uppercase.

Constants should also be static.

jovkusuma commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+
+public abstract class Task {
+
+    /**
+     * The description of the task.
+     */
```
I noticed the same issue in several other places too regarding the coding standard for variable commenting

jovkusuma commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,90 @@
+     */
+    protected String description;
+
+    /**
+     * The status of the task.
+     */
```
I noticed the same issue in several other places too regarding the coding standard for variable commenting

jovkusuma commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+
+public class Deadline extends Task {
+
+    /**
+     * The deadline of the task.
+     */
```
Not sure if this is required, but:

Javadoc of class members can be specified on a single line as follows:

```suggestion
    /** The deadline of the task. */
```

jovkusuma commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,77 @@
+
+public class Event extends Task {
+
+    /**
+     * The start date/time of the event.
+     */
```
I noticed the same issue in several other places too regarding the coding standard for variable commenting

jovkusuma commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,77 @@
+     */
+    protected LocalDateTime from;
+
+    /**
+     * The end date/time of the event.
+     */
```
I noticed the same issue in several other places too regarding the coding standard for variable commenting

jovkusuma reviewed (commented):
Overall, you did a good job following the coding standards for the most part, but try to look at the mistakes I pointed out.

PearlynnT committed:
Add Level-10

PearlynnT committed:
Edit javadoc

PearlynnT committed:
Merge branch-Level-10

PearlynnT committed:
Add varargs

PearlynnT committed:
Merge branch-A-Varargs

PearlynnT committed:
Add A-Assertions

Use assert feature to document important assumptions that should hold at various points in the code.

Add assertion to verify assumptions to todo, deadline, event, delete when the methods are invoked.

PearlynnT committed:
Add A-CodeQuality

Examine the code and refactor to improve the code quality where necessary.

Delete unnecessary comments.

PearlynnT committed:
Add A-Streams

Use the Streams feature of Java in my code.

Add Streams to find method.

PearlynnT committed:
Merge pull request #1 from PearlynnT/branch-A-Assertions

Add A-Assertions

PearlynnT committed:
Merge pull request #2 from PearlynnT/branch-A-CodeQuality

Add code quality

PearlynnT committed:
Merge pull request #3 from PearlynnT/branch-A-Streams

Add streams

PearlynnT committed:
Add A-CI

PearlynnT committed:
Add C-Sort

PearlynnT committed:
Merge pull request #4 from PearlynnT/branch-C-Sort

Add C-Sort

PearlynnT committed:
Improve GUI

PearlynnT committed:
Merge pull request #6 from PearlynnT/branch-A-BetterGui

Improve GUI

PearlynnT committed:
Add more error handling

1. Check for invalid start datetime & end datetime
2. Check for duplicate tasks

PearlynnT committed:
Fix checkstyle

PearlynnT committed:
Merge pull request #7 from PearlynnT/branch-A-MoreErrorHandling

Add more error handling

PearlynnT committed:
Add Ui.png

PearlynnT committed:
Add user guide

PearlynnT committed:
Remove command summary

PearlynnT committed:
Fix bug

PearlynnT committed:
Update README
